src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,177 @@
+    base_df2 = merge_df[merge_df['_merge'] == 'left_only'].drop(columns=['_merge']).rename(
+        columns=lambda x: x[:-2] if x.endswith('_x') else x)[list(base_df.columns)]
+    return base_df2
+
+
+# TODO
+def sync_synonyms(
+    ontology_db_path: Union[Path, str], mondo_scopes_and_xrefs_path: Union[Path, str],
+    mondo_mappings_path: Union[Path, str], onto_config_path: Union[Path, str], outpath_added: Union[Path, str],
+    outpath_confirmed: Union[Path, str], outpath_deleted: Union[Path, str], outpath_updated: Union[Path, str]
+):
+    """Create outputs for syncing synonyms between Mondo and its sources."""
+    # Iterate over mappings
+    mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    # todo: refactor filtering by prefixes
+    #  check if this pattern has been used elsewhere; I think it has. and abstract into func for these 3 lines; 'filter prefixes'
+    #  - OR use this?: get_all_owned_terms()
+    #    - it seems long and messy compared to these 3 lines, but is more comprehensive. I could make get_all_owned_terms() a simple 3 line func and then make a separate func for all edge cases?
+    # - filter by source
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mappings_df['object_prefix'] = mappings_df['object_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['object_prefix']
+
+    # Query synonyms
+    # - Mondo
+    mondo_df = pd.read_csv(mondo_scopes_and_xrefs_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    mondo_df['object_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['object_prefix']
+    mondo_df['unprefixed_source_id'] = mondo_df['source_id'].apply(lambda x: x.split(':')[1])
+    mondo_df = mondo_df[mondo_df['unprefixed_source_id'].str.isnumeric()]
+    del mondo_df['unprefixed_source_id']
+
+    # - Source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_df: pd.DataFrame = _query_synonyms(mappings_df['object_id'].tolist(), source_db)
+    source_mondo_lookup = mappings_df.set_index('object_id')['subject_id'].to_dict()
+    source_df['mondo_id'] = source_df['curie'].map(source_mondo_lookup)
+    source_df = source_df.rename(columns={'curie': 'source_id'})[['mondo_id', 'source_id', 'synonym_scope', 'synonym']]
+
+    # Determine synchronization cases
+    # TODO #3: update all on= to include 'mondo_id'?
+
+    # -confirmed
+    confirmed_df = mondo_df.merge(source_df, on=['synonym_scope', 'synonym', 'source_id'], how='inner')
+    confirmed_df['case'] = 'confirmed'
+    # TODO #1: this should have 0 rows, even if not merging on mondo_id. why does it have rows?
+    #  - Refresh mondo.sssom.tsv
+    #  example 1:
```

Comment:
## Quality issue: synonym, but no xref
I'm running into an issue where I'm starting off with `mondo.sssom.tsv` as my iterator. But I'm running into unexpected cases. I'd basically expected that: if a Mondo term has a synonym from a source, it should also have an xref to that source. But that isn't always the case.

### Example: `MONDO:0009092`
`mondo.sssom.tsv`:
```tsv
MONDO:0020749	polycystic lipomembranous osteodysplasia with sclerosing leukoencephalopathy 1	skos:exactMatch	OMIM:221770	polycystic lipomembranous osteodysplasia with sclerosing leukoencephalopathy 1	semapv:UnspecifiedMatching
```
`mondo-edit.obo`:
```obo
id: MONDO:0009092
synonym: "presenile dementia with bone cysts" EXACT [DOID:0090112, OMIM:221770]
xref: OMIMPS:221770 {source="MONDO:equivalentTo"}
# doesn't have xref for OMIM:221770. It has for OMIMPS:221770. Both exist. The synonym indeed exists on OMIM:221770.

id: MONDO:0020749
xref: OMIM:221770 {source="MONDO:equivalentTo"}   
# no synonym
```

### Questions
1. Is the above example simply caused by a typo; should `xref: OMIMPS:221770` be `xref: OMIM:221770`? Both classes do exist, FYI. The latter is a subclass of the former.
2. Are there ever situations in which you can indeed have a synonym from a source, but not have an xref to that source?
3. Is it ever allowable for there to be an xref with `{source="MONDO:equivalentTo"}`, but for there to be a synonym from that xref which is not exact?
4. What about the inverse? An xref without equivalentTo, but with an exactSynonym?

Reply:
> I'd basically expected that: if a Mondo term has a synonym from a source, it should also have an xref to that source. 
This is not the case in Mondo.

Reply:
For this task, there is no need at all to concern yourself with xrefs on synonyms. The goal of your task, in fact, is for me to delete all these xrefs and add new synonyms in.

By now, the xrefs on synonyms will be all over the place. I would not at all spend time at this stage to determine these inconsistencies.

That said: one day, we will have to. 

xref plus no synonym is totally fine, but synonym without xref is a case we need to review in 2025!

Reply:
@matentzn @twhetzel Thanks for the responses!

I'm not currently querying for xrefs on classes, so I don't need to change my implementation in light of this.

>  synonym without xref

- [ ] @matentzn Can we make sure there's a GitHub issue for this?

Reply:
@joeflack4 dont worry about this - this is part of our mega project for "removing all statements in Mondo without evidence" - right now I dont think we need specific issues for all corner cases (this is always the same: what to do with subclass axioms without evidence, synonyms without evidence, definitions without evidence, and so on).

Reply:
Alrighty; gonna mark this resolved in that case.